From: Sebastian Wick <sebastian.wick@redhat.com>
Date: Mon, 8 Sep 2025 13:19:21 +0200
Subject: brightnessManager: Handle absence of brightness control in _sync

We already call _sync in response to dimming changes, which might happen
no matter if we actually do have control over the brightness.

Let's just embrace _sync getting called in any situation and handle that
state gracefully by simply returning early. If we re-gain control, _sync
will be called again and the updated state becomes active.

Closes: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/8648

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/3864
---
 js/misc/brightnessManager.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/js/misc/brightnessManager.js b/js/misc/brightnessManager.js
index 23cd147..71f9037 100644
--- a/js/misc/brightnessManager.js
+++ b/js/misc/brightnessManager.js
@@ -175,13 +175,14 @@ export const BrightnessManager = GObject.registerClass({
             });
         }
 
-        if (this._globalScale)
-            this._sync({showOSD: false});
+        this._sync({showOSD: false});
 
         this.emit('changed');
     }
 
     _sync({showOSD = true} = {}) {
+        if (!this._globalScale)
+            return;
         if (this._inhibitUpdates)
             return;
         this._inhibitUpdates = true;
